contracts/proxy-lib/src/mutate.rs

```diff
@@ -197,6 +198,37 @@ impl ProxyContract {
             approvals.insert(signer_id);
         }
     }
+
+    pub fn update_contract(&mut self) -> Promise {
+        // Verify caller is the context config contract
+        require!(
+            env::predecessor_account_id() == self.context_config_account_id,
+            "Only the context config contract can update the proxy"
+        );
+
+        let new_code = env::input().expect("Expected proxy code");
+        // Deploy the new code and chain the callback
+        Promise::new(env::current_account_id())
+            .deploy_contract(new_code)
+            .then(Self::ext(env::current_account_id()).update_contract_callback())
+    }
+
+    #[private]
+    #[handle_result]
+    pub fn update_contract_callback(&mut self) -> Result<(), &'static str> {
+        require!(
+            env::promise_results_count() == 1,
+            "Expected 1 promise result"
+        );
+
+        match env::promise_result(0) {
+            PromiseResult::Successful(_) => {
+                env::log_str("Successfully updated proxy contract code");
+                Ok(())
+            }
+            _ => Err("Failed to update proxy contract code"),
+        }
+    }
```

Comment:
there's another callback on the context config that also just logs, do we need both?